src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -207,11 +449,121 @@ impl KVEngine for EtcdKVEngine {
         }
     }
 
+    /// Range get, return all key-value pairs start with prefix
     async fn range(&self, prefix: &KeyType) -> DatenLordResult<Vec<ValueType>> {
         let _timer = KV_METRICS.start_kv_operation_timer("range");
         let result = self.range_raw_key(prefix.to_string_key()).await?;
         Ok(result)
     }
+
+    /// Revokes a lease
+    async fn lease_revoke(&self, lease_id: i64) -> DatenLordResult<()> {
+        let mut client = self.client.clone();
+        client
+            .lease_revoke(lease_id)
+            .await
+            .with_context(|| "failed to revoke lease".to_owned())?;
+        Ok(())
+    }
+
+    /// Watch the key, return a receiver to receive the value
+    #[allow(clippy::pattern_type_mismatch)] // Raised by `tokio::select`
+    async fn watch(
+        &self,
+        prefix: &KeyType,
+    ) -> DatenLordResult<Arc<mpsc::Receiver<(String, Option<ValueType>)>>> {
+        // Create a mpsc channel, default capacity is 1024
+        let (tx, rx) = mpsc::channel(1024);
+
+        let mut client = self.client.clone();
+        // Try to watch the key prefix
+        let opt = etcd_client::WatchOptions::new().with_prefix();
+        let (mut watcher, mut watch_stream) = client
+            .watch(prefix.to_string_key(), Some(opt.clone()))
+            .await
+            .with_context(|| "Failed to create watcher".to_owned())?;
+
+        let self_prefix = prefix.to_string_key();
+
+        // Spawn a new task to handle the watch stream
+        TASK_MANAGER.spawn(TaskName::EtcdKeepAlive, |token| async move {
+            loop {
+                tokio::select! {
+                    () = token.cancelled() => {
+                        // Cancel this watch stream
+                        match watcher.cancel().await.with_context(|| "failed to cancel etcd watcher") {
+                            Ok(o) => {
+                                info!("etcd watcher task canceled, {o:?}");
+                            }
+                            Err(e) => {
+                                error!("failed to cancel etcd watcher, error={e:?}");
+                            }
+                        }
+                        break;
+                    }
+                    response = watch_stream.message() => {
+                        if let Ok(Some(watch_response)) = response {
+                            for event in watch_response.events() {
+                                // Get event data
+                                let kv = event.kv().unwrap_or_else(|| {
+                                    error!("failed to get kv from etcd watch event");
```

Comment:
1. panic is dangerouse unless you are sure the case will not happend, or etcdclient behavior has been fully tested.
2. Duplicate message with error! and panic may be unnecessary, and panic can be logged in tracing
3. If you are not sure the exact behavior of etcdclient on error condition,  define a macro to log message and panic on `cfg(debug)`, and return error on `cfg[release]` for safety on production env.

Reply:
Ok, I try to make it simple by `match None` operation.